### PR TITLE
[TRUNK-5247] Close Resources in finally block

### PR DESCRIPTION
**Pull request by BartlomiejRasztabiga**
Issue: https://issues.openmrs.org/browse/TRUNK-5247

I've closed one more Resource in finally block and changed if statement to close another resource every time. Sonar would warn otherwise.

### CODE REVIEW

File: web/src/main/java/org/openmrs/web/filter/initialization/InitializationFilter.java [hunk 1]
```diff
@@ -1192,7 +1192,7 @@ private int executeStatement(boolean silent, String user, String pw, String sql,
 		}
 		finally {
 			try {
-				if (statement != null && !statement.isClosed()) {
```
Comment (dkayiwa) [inline web/src/main/java/org/openmrs/web/filter/initialization/InitializationFilter.java]: Does this mean that statement.isClosed() is useless?

Comment (BartlomiejRasztabiga) [reply web/src/main/java/org/openmrs/web/filter/initialization/InitializationFilter.java]: I see you are closing statement on this line https://github.com/openmrs/openmrs-core/blob/48386f10175bf557801706f76c79e1413067673b/web/src/main/java/org/openmrs/web/filter/initialization/InitializationFilter.java#L1179

We can delete that and close it in finally block, so we do not duplicate code

Comment (dkayiwa) [reply web/src/main/java/org/openmrs/web/filter/initialization/InitializationFilter.java]: OK.

Comment (BartlomiejRasztabiga) [reply web/src/main/java/org/openmrs/web/filter/initialization/InitializationFilter.java]: I've pushed a fix :)

File: web/src/main/java/org/openmrs/web/filter/util/FilterUtil.java [hunk 2]
```diff
@@ -104,6 +102,14 @@ public static String restoreLocale(String username) {
 						log.debug(DATABASE_CLOSING_ERROR, e);
 					}
 				}
+
+				if (results != null) {
+					try {
+						results.close();
+					} catch (SQLException e) {
+						log.warn("Error while closing ResultSet");
```
Comment (dkayiwa) [inline web/src/main/java/org/openmrs/web/filter/util/FilterUtil.java]: Shouldn't you add the exception as the second argument in log.warn?

Comment (BartlomiejRasztabiga) [reply web/src/main/java/org/openmrs/web/filter/util/FilterUtil.java]: Done :)

**Pull request merged**